optuna/multi_objective/samplers/_motpe.py

```diff
@@ -0,0 +1,649 @@
+        trial: "multi_objective.trial.FrozenMultiObjectiveTrial",
+        config_vals: List[Optional[float]],
+        loss_vals: List[List[float]],
+    ) -> Tuple[np.ndarray, np.ndarray]:
+        """Split observations into observations for l(x) and g(x) with the ratio of gamma:1-gamma.
+
+        Weights for l(x) are also calculated in this method.
+
+        This splitting strategy consists of the following two steps:
+            1. Nondonation rank-based selection
+            2. Hypervolume subset selection problem (HSSP)-based selection
+
+        Please refer to the `original paper <https://dl.acm.org/doi/abs/10.1145/3377930.3389817>`_
+        for more details.
+        """
+        cvals = np.asarray(config_vals)
+        lvals = np.asarray(loss_vals)
+
+        # Solving HSSP for variables number of times is a waste of time.
+        # We cache the result of splitting.
+        if _SPLITCACHE_KEY in trial.system_attrs:
+            split_cache = trial.system_attrs[_SPLITCACHE_KEY]
+            indices_below = split_cache["indices_below"]
+            weights_below = split_cache["weights_below"]
+            indices_above = split_cache["indices_above"]
+        else:
+            nondomination_ranks = _calculate_nondomination_rank(lvals)
+            n_below = self._gamma(len(lvals))
+            assert 0 <= n_below <= len(lvals)
+
+            indices = np.array(range(len(lvals)))
+            indices_below = np.array([], dtype=int)
+
+            # Nondomination rank-based selection
+            i = 0
+            while len(indices_below) + sum(nondomination_ranks == i) <= n_below:
+                indices_below = np.append(indices_below, indices[nondomination_ranks == i])
+                i += 1
+
+            # Hypervolume subset selection problem (HSSP)-based selection
+            subset_size = n_below - len(indices_below)
+            if subset_size > 0:
+                rank_i_lvals = lvals[nondomination_ranks == i]
+                rank_i_indices = indices[nondomination_ranks == i]
+                worst_point = np.max(rank_i_lvals, axis=0)
+                reference_point = np.maximum(1.1 * worst_point, 0.9 * worst_point)
+                reference_point[reference_point == 0] = EPS
+                selected_indices = self._solve_hssp(
+                    rank_i_lvals, rank_i_indices, subset_size, reference_point
+                )
+                indices_below = np.append(indices_below, selected_indices)
+            assert len(indices_below) == n_below
+
+            indices_above = np.setdiff1d(indices, indices_below)
+
+            attrs = {"indices_below": indices_below, "indices_above": indices_above}
+            if self._weights is _default_weights_above:
+                weights_below = self._calculate_default_weights_below(lvals, indices_below)
+                attrs["weights_below"] = weights_below
+            study._storage.set_trial_system_attr(trial._trial_id, _SPLITCACHE_KEY, attrs)
```

Comment:
When I used SQLite as storage backend, I got the TypeError as follows:

```
[W 2020-10-13 19:27:16,311] Trial 10 failed because of the following error: TypeError('Object of type ndarray is not JSON serializable')
```

This is due to the `attrs` contains `ndarray`, so could you convert them to `List` ?

<details>
<summary>Code to reproduce the error</summary>

```python
import optuna


def objective(trial):
    # Binh and Korn function.
    x = trial.suggest_float("x", 0, 5)
    y = trial.suggest_float("y", 0, 3)

    v0 = 4 * x ** 2 + 4 * y ** 2
    v1 = (x - 5) ** 2 + (y - 5) ** 2
    return v0, v1


study = optuna.multi_objective.create_study(["minimize", "minimize"], sampler=optuna.multi_objective.samplers.MOTPEMultiObjectiveSampler(), storage="sqlite:///tmp.db")
study.optimize(objective, n_trials=100)

```
</details>